schemas/2018-05-01/deploymentTemplate.json

```diff
@@ -0,0 +1,477 @@
+            },
+            "copy": {
+              "$ref": "#/definitions/resourceCopy"
+            },
+            "comments": {
+              "type": "string"
+            }
+          }
+        }
+      ]
+    },
+    "resourceBaseExternal": {
+      "$ref": "#/definitions/resourceBase",
+      "required": [
+        "plan"
+      ]
+    },
+    "resourceSku": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the sku"
+        },
+        "tier": {
+          "type": "string",
+          "description": "Tier of the sku"
+        },
+        "size": {
+          "type": "string",
+          "description": "Size of the sku"
+        },
+        "family": {
+          "type": "string",
+          "description": "Family of the sku"
+        },
+        "capacity": {
+          "type": "integer",
+          "description": "Capacity of the sku"
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "resourceCopy": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the copy"
+        },
+        "count": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/expression"
+            },
+            {
+              "type": "integer"
+            }
+          ],
+          "description": "Count of the copy"
+        },
+        "mode": {
+          "type": "string",
+          "enum": [
+            "Parallel",
+            "Serial"
+          ],
+          "description": "The copy mode"
+        },
+        "batchSize": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/expression"
+            },
+            {
+              "type": "integer"
+            }
+          ],
+          "description": "The serial copy batch size"
+        }
+      }
+    },
+    "resourceKind": {
+      "type": "string",
+      "maxLength": 64,
+      "pattern": "(^[a-zA-Z0-9_.()-]+$)",
+      "description": "Kind of resource"
+    },
+    "resourcePlan": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the plan"
+        },
+        "promotionCode": {
+          "type": "string",
+          "description": "Plan promotion code"
+        },
+        "publisher": {
+          "type": "string",
+          "description": "Name of the publisher"
+        },
+        "product": {
+          "type": "string",
+          "description": "Name of the product"
+        },
+        "version": {
+          "type": "string",
+          "description": "Version of the product"
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "Plan of the resource"
+    },
+    "resourceLocations": {
+      "anyOf": [
+        {
+          "type": "string"
+        },
+        {
+          "enum": [
```

Comment:
we should remove the allowed values here since 1) the are not up to date and 2) don't take the cloud into consideration (should probably do that in the other schemas too)

Reply:
Not sure if anyone is using schemas from national clouds... should we just make the list complete?

Reply:
yes, tooling does use these in national clouds... I thought about a superset, but I think that would cause more problems than it solves IMO...  for example it doesn't take into account resources that have limited region availability.